packages/dev-utils/src/viem/anvil-test.ts

```diff
@@ -0,0 +1,190 @@
+import { StrongAddress } from '@celo/base'
+import { Anvil, createAnvil, CreateAnvilOptions } from '@viem/anvil'
+import {
+  Account,
+  Client,
+  createTestClient,
+  http,
+  HttpTransport,
+  PublicActions,
+  publicActions,
+  RpcSchema,
+  TestActions,
+  WalletActions,
+  walletActions,
+} from 'viem'
+import { celo, celoAlfajores } from 'viem/chains'
+import { ANVIL_PORT, DEFAULT_OWNER_ADDRESS } from '../anvil-test'
+import { TEST_BALANCE, TEST_GAS_LIMIT, TEST_GAS_PRICE, TEST_MNEMONIC } from '../test-utils'
+import { testWithViem } from './test-utils'
+
+let instance: null | Anvil = null
+
+type chains = typeof celo | typeof celoAlfajores
+export type TestClientExtended<account extends Account | undefined = Account | undefined> = Client<
+  HttpTransport,
+  chains,
+  account,
+  RpcSchema,
+  TestActions & PublicActions<HttpTransport, chains, account> & WalletActions<chains, account>
+>
+
+function createInstance(
+  stateFilePath: string,
+  chainId?: number
+): {
+  instance: Anvil
+  client: TestClientExtended
+} {
+  const port = ANVIL_PORT + (process.pid - process.ppid)
+  const options: CreateAnvilOptions = {
+    port,
+    loadState: stateFilePath,
+    mnemonic: TEST_MNEMONIC,
+    balance: TEST_BALANCE,
+    gasPrice: TEST_GAS_PRICE,
+    gasLimit: TEST_GAS_LIMIT,
+    blockBaseFeePerGas: 0,
+    stopTimeout: 1000,
+    chainId,
+  }
+
+  instance = createAnvil(options)
+
+  const client = createTestClient({
+    mode: 'anvil',
+    transport: http(),
+    chain: {
+      ...celo,
+      rpcUrls: {
+        default: { http: [`http://127.0.0.1:${instance.port}`] },
+      },
+    },
+  })
+    .extend(walletActions)
+    .extend(publicActions) as unknown as TestClientExtended
+
+  return {
+    instance,
+    client,
+  }
+}
+
+type TestWithAnvilOptions = {
+  chainId?: number
+}
+
+export function viem_testWithAnvilL1(
```

Comment:
I believe at this point testing with viem for L1 is not necessary

Reply:
which might lead to simplified code

Reply:
agreed DELETE

Reply:
agreed